objects/match.py

```diff
@@ -153,34 +154,42 @@ def getMatchData(self, censored: bool = False) -> tuple[tuple[object, int]]:
 
         # Slot mods if free mod is enabled
         if self.matchModMode == matchModModes.FREE_MOD:
-            struct.extend([(self.slots[i].mods, dataTypes.UINT32) for i in range(16)])
+            struct.extend([(slot.mods, dataTypes.UINT32) for slot in self.slots])
 
         # Seed idk
         # TODO: Implement this, it should be used for mania "random" mod
         struct.append((self.seed, dataTypes.UINT32))
 
         return tuple(struct)
 
-    def setHost(self, newHost: int) -> bool:
+    def setHost(self, newHostID: int) -> bool:
         """
         Set room host to newHost and send him host packet
 
         :param newHost: new host userID
         :return:
         """
 
-        self.remove_referee(self.hostUserID)
-        self.add_referee(newHost)
+        old_host = glob.tokens.getTokenFromUserID(self.hostUserID)
+        assert old_host is not None
 
-        slotID = self.getUserSlotID(newHost)
-        if (
-            slotID is None or
-            self.slots[slotID].user not in glob.tokens.tokens
-        ):
+        if not old_host.staff:
+            self.remove_referee(self.hostUserID)
```

Comment:
this fixes [AKAT-185](https://osuakatsuki.atlassian.net/browse/AKAT-185) - Passing host off while being staff will remove referee status